rca/api_content/content.py

```diff
@@ -0,0 +1,95 @@
+from datetime import datetime
+
+import requests
+
+
+"""
+Static methods for adding content from the live RCA api
+"""
+
+
+def parse_items_to_list(data, type):
+    items = []
+
+    for item in data["items"]:
+        _item = {}
+        if type == "News":
+            detail = item["meta"]["detail_url"] + "?fields=_,date,feed_image"
+        if type == "Event":
+            detail = item["meta"]["detail_url"] + "?fields=_,dates_times,feed_image"
+        if type == "Alumni stories":
+            detail = item["meta"]["detail_url"] + "?fields=_,feed_image,intro"
+        resp = requests.get(url=detail)
+        data = resp.json()
+        if "feed_image" in data:
+            feed_image = data["feed_image"]["meta"]["detail_url"]
+            feed_image = requests.get(url=feed_image)
+            feed_image = feed_image.json()
+            feed_image = feed_image["original"]["url"]
+            _item["image"] = feed_image
+        date = False
+        if type == "News":
+            date = data["date"]
+        if type == "Event":
+            date = data["dates_times"][0]["date_from"]
+        if type == "Alumni stories":
+            _item["description"] = data["intro"]
+        if date:
+            date = datetime.strptime(date, "%Y-%m-%d")
+            date = date.strftime("%-d %B %Y")
+            _item["description"] = date
+
+        _item["title"] = item["title"]
+        _item["type"] = type
+        _item["link"] = item["meta"]["html_url"]
+        items.append(_item)
+
+    return items
+
+
+def pull_news_and_events(programme_type_slug=None):
+    # Get the latest 2 news items and 1 event item tagged
+    # with the programme type taxonomy, if there are no events, use 3 news
+    if programme_type_slug:
+        # TODO When the api changes are merged, we will need to add the following to the query
+        # rp=programme_type_slug (https://github.com/torchbox/verdant-rca/pull/186)
+        pass
+    else:
+        # First get the events
+        # TODO extend the api to allow querying the latest event dates_times__date_from
+        # This is waiting to be merged to RCA, for now use ID
+        # eg /api/v2/pages/?limit=1&event_date_from=True&type=rca.EventItem
+        url = "https://www.rca.ac.uk/api/v2/pages/?limit=1&order=-id&type=rca.EventItem"
+        resp = requests.get(url=url)
+        data = resp.json()
+
+        if data["meta"]["total_count"] > 0:
+            news_items_to_get = 2
+            events_data = parse_items_to_list(data, "Event")
+        else:
+            news_items_to_get = 3
+
+        # News
+        url = f"https://rca.ac.uk/api/v2/pages/?limit={news_items_to_get}&order=-date&type=rca.NewsItem"
```

Comment:
```python
from django.http.request import QueryDict

query = QueryDict(mutable=True)
query.update({
    'limit': news_items_to_get,
    'order': '-date',
    'type': 'rca.NewsItem',
})
url = f"https://rca.ac.uk/api/v2/pages/?{query.urlencode}"
```
We could employ QueryDict or [urllib](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlparse) to avoid mistakes.

Reply:
Neat, thanks!